lib/eco/incparser/incparser.py

```diff
@@ -496,22 +516,27 @@ def out_of_context_analysis(self, node):
               logging.debug("OOC analysis of %s failed. Error on %s.", node, temp_parser.error_nodes)
               self.error_nodes.extend(temp_parser.error_nodes)
               node.log[("left", self.prev_version)] = saved_left
-              self.isolate(node) # XXX actually need to call refine here
-              node.isolated = True
+              temp_error = node.isolated
+              # XXX has this subtree already been isolated/refined within
```

Comment:
Does this mean "we don't have a way of telling if this subtree has been isolated/refined, so we have to do an expensive check" or ...?